src/3d/symbols/qgspointcloud3dsymbol.h

```diff
@@ -32,6 +33,27 @@
 class _3D_EXPORT QgsPointCloud3DSymbol : public QgsAbstract3DSymbol
 {
   public:
+
+    /**
+     * How to render the color of the point cloud
+     */
+    enum RenderingStyle
```

Comment:
I think it's going to be restrictive in future if we try and have a single QgsPointCloud3DSymbol class which handles all the different rendering styles. What do you think about following the same pattern as we use for the other renderers (such as vector, raster, etc) and using subclasses here instead?

Reply:
I can only think of 3 styles of rendering for now: RGB, single color and color ramp.
I think it's simpler to just have everything in one symbol. 
The advantage I see is that we can save the settings in the UI for different rendering styles and switch between them more conveniently (I think this can be achieved if we store multiple symbols and use only one but it's a bit ugly IMO).

Reply:
> I can only think of 3 styles of rendering for now: RGB, single color and color ramp.

I think we'll also need a proper "categorized" renderer (while it may be possible using a ramp, it's always going to be a hacked on UX vs something which is specifically designed for this visualisation).

Reply:
I approve, the LAS 1.4 classification categories fall in categorized while intensity or elevation are graduated

Reply:
> I think we'll also need a proper "categorized" renderer (while it may be possible using a ramp, it's always going to be a hacked on UX vs something which is specifically designed for this visualisation).

I see, I didn't think about that. I will do as you suggest @nyalldawson thanks!